### PR TITLE
Add clush-mode

### PR DESCRIPTION
**Pull request by alexvorobiev**
### Brief summary of what the package does

Emacs mode for interactive clush sessions. clush is a tool from ClusterShell framework (http://clustershell.readthedocs.io/en/latest/index.html).

### Direct link to the package repository

https://github.com/alexvorobiev/clush-mode

### Your association with the package

Author

### Relevant communications with the upstream package maintainer

None needed

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). 
- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: recipes/clush-mode [hunk 1]
```diff
@@ -0,0 +1 @@
+(clush-mode :fetcher github :repo "alexvorobiev/clush-mode")
```
Comment (DamienCassou) [inline recipes/clush-mode]: please add a new-line character at the end of the file

**Comment by DamienCassou** [comment]
Please use the linters referenced in the CONTRIBUTING.md file. You will get at least 8 warnings to fix. 

Some more feedback:

- use `set-keymap-parent` and `make-composed-keymap` instead of `nconc`
- you can use `kbd` to specify keys to add to maps
- I think you should not use backquote+quote on non-symbols in docstrings

    > e.g. "-w host" or "-g group"
- why do you use `clush-mode-hook` at the end instead of inlining `clush--initialize` inside `clush-mode`?

**Comment by purcell** [comment]
Good feedback, @DamienCassou. Couple of notes from me:

- Agree about inlining `clush--initialize`: if proper behaviour depends on that code running, it shouldn't be placed in a hook where user customization might remove it.
- It's worth adding an `;;;###autoload` cookie comment for the `clush` command.
- You're using `s` only for `s-join`, so you might consider `(mapconcat 'identity args " ")` instead.

Looks good, though, @alexvorobiev! A quick check with `package-lint` and a couple of little edits and it's ready to go. :-)

**Comment by tarsius** [comment]
I had another look and the first suggestion has not been addressed yet. I didn't look further. I am closing for now. Please let us know when things are ready so that we can reopen and do another review.

**Pull request closed without merging**